File: src/main/java/org/noxet/noxetserver/playerdata/types/PDTLong.java
package org.noxet.noxetserver.playerdata.types;

import org.bukkit.configuration.file.YamlConfiguration;
import org.noxet.noxetserver.playerdata.PlayerDataType;

public class PDTLong implements PlayerDataType<Long> {
    @Override
    public Long getEmptyValue() {
        return 0L;
    }

    @Override
    public Long getValue(YamlConfiguration config, String key) {
        return config.getLong(key);
    }
}

File: src/main/java/org/noxet/noxetserver/playerdata/types/PDTLong.scala
package org.noxet.noxetserver.playerdata.types

import org.bukkit.configuration.file.YamlConfiguration
import org.noxet.noxetserver.playerdata.PlayerDataType

class PDTLong extends PlayerDataType[Long]:
  override def getEmptyValue: Long = 0

  override def getValue(config: YamlConfiguration, key: String): Long =
    config.getLong(key)